tests/parser/FoamTests.cpp

```diff
@@ -99,11 +104,19 @@ BOOST_AUTO_TEST_CASE(FoamConfigTest) {
     BOOST_CHECK_EQUAL(fc.getRecord(0).referenceSurfactantConcentration(), 1.0);
     BOOST_CHECK_EQUAL(fc.getRecord(0).exponent(), 2.0);
     BOOST_CHECK_EQUAL(fc.getRecord(0).minimumSurfactantConcentration(), 0.3);
+    BOOST_CHECK(fc.getRecord(0).allowDesorption() == FoamData::FoamAllowDesorption::Yes);
```

Comment:
My experience is that the "sorry-path" - is actually the most important to test; so in this particular case I would say a test verifying the behavior when the `FOAMROCK` and `FOAMFSC` keywords have different length.

Reply:
Good point, will add!

Reply:
Tried to make a test with differing number of records, but then the parser got in trouble and could not parse the keyword at all. With FOAMROCK having 2 record it tried to parse "REGIONS" as the next record and refused to go on. Adding a test with missing FOAMROCK instead.